stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -442,6 +448,9 @@ public void onMessage(RespT message) {
     @Override
     public void onClose(Status status, Metadata trailers) {
       if (status.isOk()) {
+        if (!streamingResponse && unaryMessage != null) {
```

Comment:
If there is no message, it would probably be good to generate an error.

Reply:
Now it throws exception when status is OK but msg is null. A test for that is also added.